product_variant_configurator/models/pricelist.py

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Oihane Crucelaegui - AvanzOSC
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2016 ACSONE SA/NV
+# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
+
+from itertools import chain
+from odoo import api, fields, models, tools
+from odoo.exceptions import UserError
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    @api.multi
+    def _compute_price_rule(self, products_qty_partner, date=False,
+                            uom_id=False):
+        """ Low-level method - Mono pricelist, multi products
+        Returns: dict{product_id: (price, suitable_rule) for the pricelist}
+
+        If date in context: Date of the pricelist (%Y-%m-%d)
+
+            :param products_qty_partner: list of typles products, quantity,
+            partner
+            :param datetime date: validity date
+            :param ID uom_id: intermediate unit of measure
+        """
+        self.ensure_one()
+        if not date:
+            date = self._context.get('date', fields.Date.today())
+        if not uom_id and self._context.get('uom'):
+            uom_id = self._context['uom']
+        if uom_id:
+            # rebrowse with uom if given
+            # TODO: Avoid rewrite whole method for this sole line:
```

Comment:
Look to this TODO: it seems that this line is the only one that changes (and not correctly at all as I see), so I propose to rewrite it this way:
```
    @api.multi
    def _compute_price_rule(self, products_qty_partner, date=False,
                            uom_id=False):
        """Overwrite for covering the case where templates are passed and a
        different uom is used."""
        if products_qty_partner[0][0]._name != "product.template":
            # Standard use case - Nothing to do
            return self._compute_price_rule(
                products_qty_partner, date=date, uom_id=uom_id,
            )
        if not uom_id and self.env.context.get('uom'):
            uom_id = self.env.context['uom']
            ctx = self.env.context
            ctx.pop('uom')  # Remove uom context for avoiding the re-processing
            self = self.with_context(ctx)
        if uom_id:
            # rebrowse templates with uom if given
            tmpl_ids = [item[0].id for item in products_qty_partner]
            tmpl_obj = self.env['product.template']
            tmpls = tmpl_obj.with_context(uom=uom_id).browse(tmpl_ids)
            products_qty_partner = [
                (tmpls[index], data_struct[1], data_struct[2])
                for index, data_struct in enumerate(products_qty_partner)
            ]
        return self._compute_price_rule(
            products_qty_partner, date=date, uom_id=False,
        )
```